src/store/modules/product/actions.ts

```diff
@@ -145,12 +146,63 @@ const actions: ActionTree<ProductState, RootState> = {
       } else {
         showToast(translate("Products not found"));
       }
-    } catch(error) {
+    } catch (error) {
       console.error(error);
       showToast(translate("Something went wrong"));
     }
     return resp;
-  }
+  },
+
+  /**
+  * Get Product-inventory details
+  */
+  async getProductDetail({ dispatch, state }, { productId }) {
+    const current = state.current as any
+    
+    if(current && current.productId === productId) { return current }
+
+    let resp;
+    try {
+      const payload = {
+        "json": {
+          "params": {
+            "group": true,
+            "group.field": "groupId",
+            "group.limit": 10000,
+            "group.ngroups": true,
+          } as any,
+          "query": "*:*",
+          "filter": `docType: PRODUCT AND productId: ${productId}`
+        }
+      }
+      resp = await ProductService.getProductDetail(payload);
+
+      if(resp.status === 200 && resp.data.grouped.groupId?.groups.length > 0 && !hasError(resp)) {
+        let product = resp.data.grouped.groupId?.groups[0].doclist.docs[0]
+
+        product = {
+          productId: product.productId,
+          productName: product.productName,
+          brand: product.brandName,
+          externalId: product.internalName,
+          mainImage:product.mainImageUrl,
+          feature: product.productFeatures,
+          variants: product.variantProductIds
+        }
+
+        dispatch('updateCurrent', product);
```

Comment:
Why another action just for mutation? We could rename this action to updateCurrent and remove exisiting updateCurrent()

Reply:
Done Sir.